contrib/mixin/mixin.libsonnet

```
@@ -238,6 +238,20 @@
               summary: 'etcd cluster database growing very fast.',
             },
           },
+          {
+            alert: 'etcdBackendHighFragmentationRatio',
+            expr: |||
+              (etcd_mvcc_db_total_size_in_use_in_bytes/etcd_mvcc_db_total_size_in_bytes)*100 < 50
```

use of min_over_time() will account for transient failed scrapes.
```suggestion
              (min_over_time(etcd_mvcc_db_total_size_in_use_in_bytes[5m]) / min_over_time(etcd_mvcc_db_total_size_in_bytes[5m])) < 0.5
```

---

Will `'for': '10m'` workaround the transient failed scrape? 

Usually both `etcd_mvcc_db_total_size_in_use_in_bytes` and `etcd_mvcc_db_total_size_in_bytes` are monotonic increasing for KEY space unless there are some compaction & de-fragmentation operations. So the latest value or `max_over_time` makes more sense than `min_over_time`?

---

> Will `'for': '10m'` workaround the transient failed scrape?

No. In case of a failed scrape, Prometheus will insert stale markers and the alert expression will return no data right after the failed scrape which will resolve the alert. So you'll see:
* alert firing
* failed scrape -> alert resovled
* scrape ok -> alert pending
* ...
* 10 minutes elapsed -> alert firing

> Usually both `etcd_mvcc_db_total_size_in_use_in_bytes` and `etcd_mvcc_db_total_size_in_bytes` are monotonic increasing for KEY space unless there are some compaction & de-fragmentation operations. So the latest value or `max_over_time` makes more sense than `min_over_time`?

in fact `last_over_time()` would be better in this case.

---

OK, then why not just `(etcd_mvcc_db_total_size_in_use_in_bytes/etcd_mvcc_db_total_size_in_bytes)*100 < 50 `instead of `(last_over_time(etcd_mvcc_db_total_size_in_use_in_bytes[5m]) / last_over_time(etcd_mvcc_db_total_size_in_bytes[5m])) < 0.5`?

---

It's an interesting & subtle discussion :). 

I am thinking should we use `last_over_time` for all similar cases (see an example below)? Which of course can be resolved in a separate PR if needed. 
```
(etcd_mvcc_db_total_size_in_bytes/etcd_server_quota_backend_bytes)*100 > 95
```

---

@ahrtr sure thing, will raise a fix following this one 👍🏽

---

@ahrtr I have created https://github.com/etcd-io/etcd/pull/14237 to address your request above